File: 0137-single-number-ii/0137-single-number-ii.py
class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        
        nums.sort()
        
        boo = False
        
        for i in range(1,len(nums)):
            
            cur, pre = nums[i], nums[i-1]
            
            if pre == cur:
                boo = True
                
            else:
                if boo == True:
                    boo = False
                else:
                    return pre
                
        if boo == False:
            return nums[-1]
            

File: 0137-single-number-ii/0137-single-number-ii.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        
        unordered_set<long> set;
        long sumNum = 0;
        
        // convert vector to set
        for (int num : nums){
            set.insert((long)num);
            sumNum += num;
        }
        
        // calculate sum
        long sumSet = 0;
        for (int num:set){
            sumSet += num;
        }
        
        // res = (sumSet*3-sumNum)/2
        return (int)((sumSet*3-sumNum)/2);
    }
};